olp-cpp-sdk-core/include/olp/core/cache/DefaultCache.h

```diff
@@ -62,6 +62,7 @@ class CORE_API DefaultCache : public KeyValueCache {
   enum StorageOpenResult {
     Success,             /*!< The operation succeeded. */
     OpenDiskPathFailure, /*!< The disk cache failure. */
+    CacheCorrupted,      /*!< The disk cache is corrupted. */
```

Comment:
I think we should differentiate here. I would name it `ProtectedCacheCorrupted` as we are not interested in mutable cache corruption as that one can heal itself without any issues.